### PR TITLE
applications: nrf_desktop: Fetching data through dongle

### PR DESCRIPTION
**Pull request by Qbicz**
Forward fetch requests and forward fetched data on config channel.

Fixed transaction state management when forwarding.
Fixed not resetting timeout when new transaction was starting.
Added filtering for fetch requests by transport (USB/BLE). Fetched data
is now only processed on the correct transport (important for mouse
which has both USB and BLE).

Jira: DESK-564

Signed-off-by: Filip Kubicz <filip.kubicz@nordicsemi.no>

### CODE REVIEW

**Comment by Qbicz** [comment]
@pdunaj @MarekPieta can you review?

File: applications/nrf_desktop/src/modules/hid_forward.c [hunk 1]
```diff
@@ -306,6 +357,42 @@ static bool event_handler(const struct event_header *eh)
 
 			return false;
 		}
+
+		if (is_config_forward_get_event(eh)) {
+
+			if (forward_pending) {
+				LOG_INF("GATT read already pending");
+				 return false;
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/modules/hid_forward.c]: fix indentation

File: applications/nrf_desktop/src/modules/hid_forward.c [hunk 2]
```diff
@@ -170,6 +173,49 @@ void hidc_write_cb(struct bt_gatt_hids_c *hidc,
...
+		int pos = config_channel_report_parse(data, REPORT_SIZE_USER_CONFIG,
+						      &frame, true);
+		if (pos < 0) {
+			LOG_WRN("Could not set report");
+			return pos;
+		}
+
+		if ((frame.status != CONFIG_STATUS_SUCCESS) ||
+		    (frame.event_data_len == 0)) {
+			LOG_INF("GATT read done, but fetch was not ready yet.");
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/modules/hid_forward.c]: Do we need dot at the end ?

File: applications/nrf_desktop/src/events/config_event.h [hunk 3]
```diff
@@ -97,6 +97,7 @@ struct config_fetch_event {
 
 	u16_t recipient;
 	u8_t id;
+	bool usb;
```
Comment (pdunaj) [inline applications/nrf_desktop/src/events/config_event.h]: please use "void *channel_id" and pass address to config_channel instance - it's more transparent

File: applications/nrf_desktop/src/modules/hid_forward.c [hunk 4]
```diff
@@ -170,6 +173,49 @@ void hidc_write_cb(struct bt_gatt_hids_c *hidc,
...
+
+		struct config_fetch_event *event =
+			new_config_fetch_event(frame.event_data_len);
+
+		event->id = frame.event_id;
+		event->recipient = frame.recipient;
+		event->usb = true;
+
+		k_spinlock_key_t key = k_spin_lock(&lock);
+		memcpy(event->dyndata.data, &data[pos], frame.event_data_len);
```
Comment (pdunaj) [inline applications/nrf_desktop/src/modules/hid_forward.c]: this is expensive? why spinlock?

File: applications/nrf_desktop/src/modules/hid_forward.c (hunk 1 above)
Comment (Qbicz) [reply applications/nrf_desktop/src/modules/hid_forward.c]: done

File: applications/nrf_desktop/src/modules/hid_forward.c (hunk 2 above)
Comment (Qbicz) [reply applications/nrf_desktop/src/modules/hid_forward.c]: done

File: applications/nrf_desktop/src/events/config_event.h (hunk 3 above)
Comment (Qbicz) [reply applications/nrf_desktop/src/events/config_event.h]: done

File: applications/nrf_desktop/src/modules/hid_forward.c (hunk 4 above)
Comment (Qbicz) [reply applications/nrf_desktop/src/modules/hid_forward.c]: done

**Comment by Qbicz** [comment]
Rebased, resolved conflict. Removed usb param because void *channel_id was used instead.

Tested with `fwinfo` command.

@pdunaj can you review?

**Comment by Qbicz** [comment]
CI passed. @pizi-nordic can you merge?

**Pull request merged**